src/app/context/SettingsContext.tsx

```diff
@@ -55,19 +60,41 @@ export const SettingsProvider = ({
       });
   }, []);
 
-  // update rate
-  useEffect(() => {
-    api.getCurrencyRate().then((response) => {
-      setCurrencyRate(response.rate);
-    });
-  }, [settings.currency]);
-
-  const getFiatValue = (amount: number | string) =>
-    getFiatValueFunc({
-      amount,
-      rate: currencyRate,
-      currency: settings.currency,
-    });
+  const getCurrencyRate = async (): Promise<number> => {
+    // ensure to get the correct rate for current currency in state
+    if (settings.currency !== currencyRate.current?.currency) {
+      const response = await api.getCurrencyRate(); // gets rate from browser.storage or API
+
+      // update local ref
+      currencyRate.current = {
+        rate: response.rate,
+        currency: settings.currency,
+      };
+    }
+
+    return currencyRate.current.rate;
+  };
+
+  const getFiatValue = async (amount: number | string) => {
+    try {
+      const rate = await getCurrencyRate();
+      const value = getFiatValueUtil({
+        amount,
+        rate,
+        currency: settings.currency,
+      });
+
+      return value;
+    } catch (e) {
+      console.error(e);
+      if (e instanceof Error)
+        toast.error(
+          `SettingsProvider: getFiatValue with currency ${settings.currency} failed. (${e.message})`
+        );
+
+      return "Fiat Error";
```

Comment:
maybe we only return an empty string? otherwise we print "Fiat Error" everywhere - which is also not very descriptive for the user.

Reply:
Fine for me. But when the toast are removed now, at least something should be shown to the user that there was an error.

Changed it now to "??"

Reply:
that error should be an exception, so I'd not optimize for that error. and for the user it's obvious that there is something wrong when the fiat amount is not show.